src/form/input/file/file-unput.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import {render, fireEvent} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+
+import {testA11y} from "../../../core/utils/test/testUtils";
+import FileInput, {FileInputProps} from "./FileInput";
+
+describe("<FileInput />", () => {
+  const defaultFileInputProps: FileInputProps = {
+    testid: "file-input",
+    name: "file-input",
+    htmlFor: "file-input",
+    onChange: jest.fn(),
+    children: "Upload File"
+  };
+
+  it("should render correctly", () => {
+    render(<FileInput {...defaultFileInputProps} />);
+  });
+
+  it("should match snapshot", () => {
+    const tree = create(<FileInput {...defaultFileInputProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<FileInput {...defaultFileInputProps} />);
+
+    await testA11y(container);
+  });
+
+  it("should render children correctly", () => {
+    const {getByTestId} = render(<FileInput {...defaultFileInputProps} />);
+
+    expect(getByTestId(`${defaultFileInputProps.testid}.label`)).toHaveTextContent(
+      "Upload File"
+    );
+  });
+
+  it("should upload file correctly", () => {
+    const file = new File(["hipo"], "hipo.png", {type: "image/png"});
+
+    render(<FileInput {...defaultFileInputProps} />);
+    const fileInput = document.getElementsByTagName("input")[0]!;
+
+    fireEvent.change(fileInput, {target: {files: [file]}});
+
+    expect(fileInput.files![0]).toEqual(file);
+    expect(defaultFileInputProps.onChange).toHaveBeenCalledTimes(1);
+  });
+
+  it("should upload multiple files correctly", () => {
+    const files = [
+      new File(["hipo"], "hipo.png", {type: "image/png"}),
+      new File(["labs"], "labs.png", {type: "image/png"})
+    ];
+
+    render(<FileInput {...defaultFileInputProps} />);
+
+    const fileInput = document.getElementsByTagName("input")[0]!;
+
+    fireEvent.change(fileInput, {target: {files}});
+
+    expect(fileInput.files).toHaveLength(files.length);
+    expect(fileInput.files![0]).toStrictEqual(files[0]);
+    expect(fileInput.files![1]).toStrictEqual(files[1]);
+    expect(defaultFileInputProps.onChange).toHaveBeenCalledTimes(files.length);
+  });
+

```

Comment:
As far as I can see, in these test cases, we are basically testing native `input` element. (Because we select input tag and fire event in this element) Since it is native browser element, testing seems redundant. I believe these can be omitted.

Reply:
Makes sense. I removed this tests. Thank you